test/parallel/test-async-wrap-constructor.js

```diff
@@ -0,0 +1,32 @@
+'use strict'
+const common = require('../common');
+if (!common.hasCrypto) {
+  common.skip('missing crypto');
+  return;
+}
+const assert = require('assert');
+const async_hooks = require('async_hooks');
+
+try {
+  async_hooks.createHook({ init: 1 })
+} catch (err) {
+  assert.equal(err.message, 'init must be a function')
+}
```

Comment:
Can you re-phrase these using `assert.throws`? e.g.

```js
assert.throws(() => {
  async_hooks.createHook({ init: 1 });
}, /^TypeError: init must be a function$/);
```

(Also, please make sure that `make lint` passes :))